src/app/home/todos/todos.component.ts

```diff
@@ -47,28 +59,44 @@ export class TodosComponent implements OnInit, OnDestroy {
   }
 
   get filterValue(): TodoFilter {
-    if (typeof window !== 'undefined') {
-      return (window.location.hash.replace(/^#\//, '') as TodoFilter) || TodoFilter.All;
+    if (typeof window !== "undefined") {
+      return (
+        (window.location.hash.replace(/^#\//, "") as TodoFilter) ||
+        TodoFilter.All
+      );
     }
     return TodoFilter.All;
   }
 
   get itemsLeft() {
-    return (this.todos || []).filter(t => !t.completed).length;
+    return (this.todos || []).filter((t) => !t.completed).length;
   }
 
   clearCompleted() {
-    (this.todos || []).filter(t => t.completed).forEach(t => this.delete.emit(t));
+    (this.todos || [])
+      .filter((t) => t.completed)
+      .forEach((t) => this.delete.emit(t));
+
+    this.todos.splice(0, this.todos.length);
+  }
+
+  placeHolderMessageText() {
+    return (this.name = "What needs to be done?");
   }
 
   addTodo(input: HTMLInputElement) {
-    const todo = {
-      completed: false,
-      label: input.value,
-    };
-    const result: Todo = { ...todo, id: Math.random().toString() };
-    this.todos.push(result);
-    input.value = '';
+    if (input.value == "" || input.value == null || input.value == undefined) {
+      this.name = "Please Write Something!";
```

Comment:
The placeholder is not the right place to display an error. We should either show a popup, or an error field somewhere.

Reply:
There needs to be analysis, so there is a lot to do.